Sources/BraintreeCoreSwift/BTAPIClientSwift.swift

```diff
@@ -1,5 +1,441 @@
 import Foundation
 
+/// This class acts as the entry point for accessing the Braintree APIs via common HTTP methods performed on API endpoints.
+/// - Note: It also manages authentication via tokenization key and provides access to a merchant's gateway configuration.
+// TODO: rename once Objective-C files are deleted
 @objcMembers public class BTAPIClientSwift: NSObject {
-    
+
+    public typealias RequestCompletion = (BTJSON?, HTTPURLResponse?, Error?) -> Void
+
+    // MARK: - Internal Properties
+
+    var tokenizationKey: String?
+    var clientToken: BTClientToken?
+
+    /// Client metadata that is used for tracking the client session
+    var metadata: BTClientMetadata?
+
+    /// Used to fetch and store configurations in the URL Cache of the session
+    var configurationHTTP: BTHTTP?
+
+    var http: BTHTTP?
+    var apiHTTP: BTAPIHTTP?
+    var graphQLHTTP: BTGraphQLHTTP?
```

Comment:
Right now they need to be optional since we don't know the values coming into this class for any of these and this class essentially acts as a pass-thru to BTHTTP depending on what is set. 😕 Making them optional requires us to rewrite this networking layer from scratch, which we _should_ do at some point but probably not in this PR/major version.